src/app/index.ts

```diff
@@ -1,15 +1,17 @@
+import h from "hyperscript"
 import { Explorer } from "~explorer"
 import { NavigationBar } from "~navigation-bar"
 
 import { rootFolder, setCurrentFolder } from "./state"
-
 import "./app.css"
 
-export const appElement = document.getElementById("app") as HTMLElement
+export let appElement: HTMLElement | null = null
 
-export function App() {
-  NavigationBar()
-  Explorer()
+export function App(rootElement: HTMLElement) {
+  appElement = h<HTMLDivElement>("div.app")
+  rootElement.appendChild(appElement)
+  appElement.appendChild(NavigationBar(appElement))
+  appElement.appendChild(Explorer(appElement))
```

Comment:
I started passing `appElement` directly as a param here so we are always assured, `appElement` always exists within children and we won't have to do `appElement?.addEventListener`